tutorials/optimization/optimizing_3d_performance.rst

```diff
@@ -0,0 +1,143 @@
+.. meta::
+    :keywords: optimization
+
+.. _doc_optimizing_3d_performance:
+
+Optimizing 3D performance
+=========================
+
+Culling
+=======
+
+Godot will automatically perform view frustum culling in order to prevent
+rendering objects that are outside the viewport. This works well for games that
+take place in a small area, however things can quickly become problematic in
+larger levels.
+
+Occlusion culling
+~~~~~~~~~~~~~~~~~
+
+Walking around a town for example, you may only be able to see a few buildings
+in the street you are in, as well as the sky and a few birds flying overhead. As
+far as a naive renderer is concerned however, you can still see the entire town.
+It won't just render the buildings in front of you, it will render the street
+behind that, with the people on that street, the buildings behind that. You
+quickly end up in situations where you are attempting to render 10x, or 100x
+more than what is visible.
+
+Things aren't quite as bad as they seem, because the Z-buffer usually only
+allows the GPU to fully shade the objects that are at the front. However,
```

Comment:
z-buffer usually allows the GPU to fully shade only the objects that are at the front